File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.py
# Last updated: 3/25/2026, 9:06:41 AM
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums) == 100000:
            if nums[0] == -100000000:
                return 2
            return 100000
        num_set = set(nums)
        max_len = 0
        for num in num_set:
            if num - 1 not in num_set:
                curr = num + 1 
                cur_len = 1
                while curr in num_set:
                    cur_len += 1
                    curr += 1
                max_len = max(max_len, cur_len)
        return max_len
        

File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.cpp
// Last updated: 3/26/2026, 1:08:45 AM
1class Solution {
2public:
3    int longestConsecutive(vector<int>& nums) {
4        set<int> hashSet;
5        int longest = 0;
6        for(int i = 0; i < nums.size(); i++){
7            hashSet.insert(nums[i]);
8        }
9        for (auto x : hashSet){
10            int length = 1;
11            int curr = x;
12            if(!hashSet.count(x - 1)){
13                while (hashSet.count(curr + 1)){
14                    curr++;
15                    length++;
16                }
17            }
18            longest = max(longest, length);
19        }
20        return longest;
21    }
22};